### PR TITLE
Snapshot exclusion

### LINKED ISSUE #117: Automatic snapshotting of Dyn tables should be supressed
Seen on CPRO Staging  
    {
        "logTime": "19.01.2017 11:29:45.148963",
        "logLevel": "info",
        "pid": "<0.2269.0>",
        "module": "imem_snap",
        "function": "handle_info",
        "line": 0,
        "node": "'cpro@ZHHAPMOP-SBSM01.it.bwns.ch'",
        "fields": "[snapshot]",
        "message": "snapshot created for cproNodeJobDyn@ZHHAPMOPSBSM01",
        "stacktrace": "[]"
    }

### LINKED ISSUE #119: Expose functions to exclude tables from automatic snapshoting
The idea is to be able to dynamically add or remove tables to the exclusion list accessible from other applications when transient tables are created.

### PR DESCRIPTION
**Pull request by shamis**
Fixes #119 
Fixes #117

### CODE REVIEW

File: src/imem_snap.erl [hunk 1]
```diff
@@ -57,16 +58,18 @@
...
+<<"fun(ExcludePatterns) ->
     ExcludeList = [dual, ddSize, ddNode
                   ,imem_meta:physical_table_name(ddCache@)
                   ,imem_meta:physical_table_name(ddSeCo@)
                   ,imem_meta:physical_table_name(ddPerm@)
                   ,imem_meta:physical_table_name(mproConnectionProbe@)
                   ],
-    ExcludeSuff = [\"Dyn\",\"Audit\",\"Hist\",\"Idx\"],
-    ExcludePred = fun(AN) -> SN = atom_to_list(AN), (lists:usort([lists:suffix(E,SN) || E <- ExcludeSuff]) == [false]) end,
+    ExcludePred = fun(AN) -> SN = atom_to_list(AN), (lists:usort([re:run(SN, E) || E <- ExcludePatterns]) == [nomatch]) end,
```
Comment (c-bik) [inline src/imem_snap.erl]: an optimal alternative way to write this is...
```erlang
ExcludePred = fun(AN) ->
    SN = atom_to_list(AN),
    [error || E <- ExcludePatterns], re:run(SN, E) /= nomatch) == []
end
```

File: src/imem_snap.erl [hunk 2]
```diff
@@ -23,6 +23,7 @@
         , restore_chunked/3
         , del_dirtree/1
         , snap_file_count/0
+        , exclude_table_pattern/1
```
Comment (c-bik) [inline src/imem_snap.erl]: why export?

Comment (shamis) [inline src/imem_snap.erl]: because of this https://github.com/K2InformaticsGmbH/imem/issues/119 requirement

File: src/imem_snap.erl [hunk 3]
```diff
@@ -53,47 +54,70 @@
...
+-define(PUT_SNAPSHOT_EXCLUSION_PATTERNS(__TablePatterns, __Remark), 
+            ?PUT_CONFIG(snapshotExclusuionPatterns, [], __TablePatterns, __Remark)).
+-define(GET_SNAPSHOT_SCRIPT_FUN,
+            ?GET_CONFIG(snapshotScriptFun, [],
+                        <<"fun(ExcludePatterns) ->
+                            ExcludeList = [dual, ddSize, ddNode
+                                          ,imem_meta:physical_table_name(ddCache@)
+                                          ,imem_meta:physical_table_name(ddSeCo@)
+                                          ,imem_meta:physical_table_name(ddPerm@)
+                                          ,imem_meta:physical_table_name(mproConnectionProbe@)
```
Comment (c-bik) [inline src/imem_snap.erl]:
```erlang
[dual, ddSize, ddNode | [imem_meta:physical_table_name(T) || T <- [ddCache@,ddSeCo@,ddPerm@]]
```
`mproConnectionProbe@` isn't a IMEM table

Comment (shamis) [inline src/imem_snap.erl]: `mproConnectionProbe@` was not added in this PR

**Pull request merged**